src/com/oltpbenchmark/benchmarks/tpcc/procedures/NewOrder.java

```diff
@@ -101,35 +85,77 @@
 	private PreparedStatement stmtInsertNewOrder = null;
 	private PreparedStatement stmtUpdateDist = null;
 	private PreparedStatement stmtInsertOOrder = null;
-	private PreparedStatement stmtGetItem = null;
-	private PreparedStatement stmtGetStock = null;
 	private PreparedStatement stmtUpdateStock = null;
 	private PreparedStatement stmtInsertOrderLine = null;
+	private PreparedStatement stmtGetItemsAndStock = null;
+
+	public NewOrder() {
+		stmtGetItemsAndStockSQL = new SQLStmt[15];
+		stmtUpdateStockSQL = new SQLStmt[15];
+		stmtInsertOrderLineSQL = new SQLStmt[11];
+
+		StringBuilder sb = new StringBuilder();
+		sb.append("SELECT I_ID, I_PRICE, I_NAME , I_DATA, ");
+        sb.append("S_W_ID, S_QUANTITY, S_DATA, S_DIST_01, S_DIST_02, S_DIST_03, S_DIST_04, S_DIST_05, S_DIST_06, S_DIST_07, S_DIST_08, S_DIST_09, S_DIST_10 FROM ");
+		sb.append(TPCCConstants.TABLENAME_ITEM);
+		sb.append(", ");
+		sb.append(TPCCConstants.TABLENAME_STOCK);
+		sb.append(" WHERE S_I_ID = I_ID AND S_W_ID = ? AND S_I_ID in (");
+		for (int ii = 1; ii <= 15; ++ii) {
+			if (ii == 1) sb.append("?");
+			else sb.append(",?");
+			stmtGetItemsAndStockSQL[ii - 1] = new SQLStmt(sb.toString() + ") FOR UPDATE");
+		}
 
+		sb = new StringBuilder();
+		sb.append("UPDATE ");
+		sb.append(TPCCConstants.TABLENAME_STOCK);
+		sb.append(" SET S_QUANTITY = r.q, S_YTD = S_YTD + r.y, S_ORDER_CNT = S_ORDER_CNT + 1,  S_REMOTE_CNT = S_REMOTE_CNT + r.c ");
+		sb.append(" FROM (SELECT ");
+
+		StringBuilder sb2 = new StringBuilder();
+		sb2.append("UNNEST(ARRAY[");
+
+		for (int ii = 1; ii <= 15; ++ii) {
+			if (ii == 1) sb2.append("?");
+			else sb2.append(", ?");
+
+			StringBuilder sb3 = new StringBuilder();
+			sb3.append(sb);
+			sb3.append(sb2);
+			sb3.append("]) as i, ");
+			sb3.append(sb2);
+			sb3.append("]) as q, ");
+			sb3.append(sb2);
+			sb3.append("]) as y, ");
+			sb3.append(sb2);
+			sb3.append("]) as c");
+			sb3.append(")r");
+			sb3.append(" WHERE STOCK.S_I_ID = r.i AND STOCK.S_W_ID = ?");
```

Comment:
might be cleaner to use something like this: https://docs.oracle.com/javase/7/docs/api/java/util/Formatter.html